### ISSUE TITLE
Warlock pet not spawning after dismount

### ISSUE THREAD

**Issue by vildheden** [chromiecraft/chromiecraft#2852]
### What client do you play on?

enUS

### Faction

- [ ] Alliance
- [X] Horde

### Content Phase:

- [ ] Generic
- [ ] 1-19
- [X] 20-29
- [ ] 30-39
- [ ] 40-49
- [ ] 50-59
- [ ] 60

### Current Behaviour

I get random episodes where my pet wont spawn after I dismount so have to resummon it over and over.
Had 2 periods today where it didnt work first was in wsg and second was just in the open world - then suddenly out of nowhere it started working again. Then I went for dinner and came back on and again it does not work.

### Expected Blizzlike Behaviour

Pet should spawn if it is summoned before u use mount and then dismount - works sometimes.

### Steps to reproduce the problem

I cant reproduce - seems random

### AC rev. hash/commit

https://github.com/chromiecraft/azerothcore-wotlk/commit/259b9133f68ef0d740fc871d59fab3d2791f33b4

### Operating system

Windows 10

### Modules

- [mod-ah-bot](https://github.com/azerothcore/mod-ah-bot)
- [mod-cfbg](https://github.com/azerothcore/mod-cfbg)
- [mod-chat-transmitter](https://github.com/azerothcore/mod-chat-transmitter)
- [mod-chromie-xp](https://github.com/azerothcore/mod-chromie-xp)
- [mod-desertion-warnings](https://github.com/azerothcore/mod-desertion-warnings)
- [mod-duel-reset](https://github.com/azerothcore/mod-duel-reset)
- [mod-eluna-lua-engine](https://github.com/azerothcore/mod-eluna-lua-engine)
- [mod-ip-tracker](https://github.com/azerothcore/mod-ip-tracker)
- [mod-low-level-arena](https://github.com/azerothcore/mod-low-level-arena)
- [mod-multi-client-check](https://github.com/azerothcore/mod-multi-client-check)
- [mod-pvp-titles](https://github.com/azerothcore/mod-pvp-titles)
- [mod-pvpstats-announcer](https://github.com/azerothcore/mod-pvpstats-announcer)
- [mod-queue-list-cache](https://github.com/azerothcore/mod-queue-list-cache)
- [mod-server-auto-shutdown](https://github.com/azerothcore/mod-server-auto-shutdown)
- [mod-save-inventory](https://github.com/Nyeriah/mod-save-inventory)
- [lua-carbon-copy](https://github.com/55Honey/Acore_CarbonCopy)
- [lua-exchange-npc](https://github.com/55Honey/Acore_ExchangeNpc)
- [lua-custom-worldboss](https://github.com/55Honey/Acore_CustomWorldboss)
- [lua-level-up-reward](https://github.com/55Honey/Acore_LevelUpReward)
- [lua-recruit-a-friend](https://github.com/55Honey/Acore_RecruitAFriend)
- [lua-send-and-bind](https://github.com/55Honey/Acore_SendAndBind)
- [lua-temp-announcements](https://github.com/55Honey/Acore_TempAnnouncements)
- [lua-zonecheck](https://github.com/55Honey/acore_Zonecheck)
- [lua-zone-debuff](https://github.com/55Honey/Acore_ZoneDebuff)

### Customizations

None

### Server

ChromieCraft

**Comment by ishidinho** [comment]
Hey @vildheden. Thank you for the report.

Sadly, I was was not able to reproduce the problem.

I did this for 5 minutes. Hab no problems.

https://user-images.githubusercontent.com/62797536/150786059-e8f09a36-9a3c-4435-b43a-db6be02a3aae.mp4

I think we need some clear circumstances here, when the issue appears.

**Comment by vildheden** [comment]
https://user-images.githubusercontent.com/98283901/150943292-2566d5f1-7790-4478-8f5d-068eddfd52c1.mp4

It happened again today when i logged on, but after a reboot of the game it was back to normal

**Comment by vildheden** [comment]
https://user-images.githubusercontent.com/98283901/150984830-a3e1df29-b9f3-4863-8d2f-77390910e6ab.mp4

This time it was working and then after I did SM Armory and it stopped working again..

**Comment by vildheden** [comment]
Logged in last night and it didnt work again.. Tried to disable addons but then it still didnt work.. Enabled them again and after 2 game reboots it was working again..
I have tried on 2 different PC's and im using ur download of the game..

**Comment by vildheden** [comment]
https://user-images.githubusercontent.com/98283901/151339540-ed1c3505-2b24-4a3e-809c-96dc5ba1e916.mp4

Same issue today on PC that does not have the HD pack..

**Comment by Adrialla1** [comment]
Same issue with my 36 Horde Warlock. Demon despawns as I mount (as it should) but does not respawn back when I dismount. No shard is returned, either. Rebooting didn't help. I have HD. I hadn't played in a few weeks. It worked fine the last time I played. When I logged in today it was broken. No changes have been made to the files/downloads/addons.

**Comment by dakootagit** [comment]
Confirmed. Live server. 

 I do use Addons and have not tested without them. Warlock pet disappears when mounting. I’ve notice relogging sometimes causes the issue to stop but often later occurs. Changing talents also seems too make the issues stop. I do not have dual spec so not sure if toggling would be another fix. 
 
Demonology seems to have the issue occur the most then affliction and lastly destruction. 

In battle grounds pets sometimes won’t spawn which is an indicator the mount is also bugged. When the pets do spawn…first spawn is half HP half mana…each time after 1%hp 1%mana(the pet looks like it’s spawning from the ground like a DK pet when this happens) this remains until a hunter kills the pet and gets his revenge for all the dots on his. 

TLDR: Confirmed. this still occurs

**Comment by dakootagit** [comment]
I also noticed something odd getting it to stop.

was affliction pet bug occurred 
Went demo
Farmed for a little bit
Mounted pet bug
Logged out
Logged in
Same bug
Did this 3 more times(tired of respecting)
On the 3rd log out my pet appeared on the character screen
Logged in
Now it works
TLDR: don’t fix this just give all warlocks dual spec for free. Currently trying to farm the gold to test with dual spec.

**Comment by vildheden** [comment]
Now usually when it happens I keep log out untill my demon shows up on character screen then it works after that..

**Comment by acidmanifesto** [comment]
Issue marked as confirmed but still under evaulation and still being looked into by the developers.

**Comment by Nefertumm** [comment]
Closed by https://github.com/azerothcore/azerothcore-wotlk/commit/acd3ed8759477e6059b31f88377a790f3a0d8133
